services/app-web/src/hooks/useDebouncedMutation.ts

```diff
@@ -0,0 +1,61 @@
+import {
+    ApolloCache,
+    DefaultContext,
+    DocumentNode,
+    MutationFunctionOptions,
+    MutationHookOptions,
+    MutationTuple,
+    TypedDocumentNode,
+    useMutation,
+} from '@apollo/client'
+
+import { debounce } from '../common-code/debounce/debounce'
+
+// mostly from https://stackoverflow.com/a/74109414/5351022
+export const useDebouncedMutation = <
```

Comment:
This hook fixed the issue. Should we put a comment here about why we are debouncing mutations to not cause confusion? Not sure if it's just me, but I usually associate debounces with inputs and not mutations.